packages/devtools_app/test/memory/diff/widgets/class_filter_test.dart

```diff
@@ -13,20 +14,42 @@ import 'package:flutter_test/flutter_test.dart';
 import '../../../matchers/matchers.dart';
 import '../../../scenes/memory/diff_snapshot.dart';
 
+final _customFilter =
+    ClassFilter(filterType: ClassFilterType.only, except: '', only: '');
+
+class _FilterTest {
+  _FilterTest(this.isDiff);
+
+  final bool isDiff;
+  String get name => isDiff ? 'diff' : 'single';
+}
+
+final _tests = [_FilterTest(false), _FilterTest(true)];
+
 void main() {
   late DiffSnapshotScene scene;
 
-  Future<void> pumpScene(WidgetTester tester) async {
-    await tester.pumpWidget(scene.build());
-    await expectLater(
-      find.byType(SnapshotInstanceItemPane),
-      matchesDevToolsGolden('../../../goldens/memory_diff_snapshot_scene.png'),
-    );
+  Future<void> pumpScene(WidgetTester tester, _FilterTest test) async {
     expect(
       scene.diffController.core.snapshots.value
           .where((element) => element.hasData),
       hasLength(2),
     );
+
+    if (test.isDiff) {
+      scene.diffController.setDiffing(
+        scene.diffController.derived.selectedItem.value as SnapshotInstanceItem,
+        scene.diffController.core.snapshots.value[1] as SnapshotInstanceItem,
+      );
+    }
+
+    await tester.pumpWidget(scene.build());
+    await expectLater(
+      find.byType(SnapshotInstanceItemPane),
+      matchesDevToolsGolden(
+        '../../../goldens/memory_diff_snapshot_scene_${test.name}.png',
```

Comment:
```suggestion
        test.path,
```
Might be worth generating the path in _FilterTest